crates/cairo-lint-core/tests/test_files/equatable_if_let/equatable_if_let

```diff
@@ -0,0 +1,261 @@
+    }
+}
+
+//! > ==========================================================================
+
+//! > complex equality destructuring if let
+
+//! > cairo_code
+fn main() {
+    let x = Option::Some(1_felt252);
+    let y = Option::Some(1_felt252);
+    if let (Option::Some(a), Option::Some(b)) = (x, y) {
+        if a == b {
+            println!("{a}");
+        }
+    }
+}
+
+//! > diagnostics
+
+//! > fixed
+fn main() {
+    let x = Option::Some(1_felt252);
+    let y = Option::Some(1_felt252);
+    if let (Option::Some(a), Option::Some(b)) = (x, y) {
+        if a == b {
+            println!("{a}");
+        }
+    }
+}
+
+//! > ==========================================================================
+
+//! > Simple Value Pattern Matching
+
+//! > cairo_code
+fn main() {
+    let a = 2;
+    if let 2 = a {}
+}
+
+//! > diagnostics
+warning: Plugin diagnostic: `if let` pattern used for equatable value. Consider using a simple comparison `==` instead
+ --> lib.cairo:4:5
+  |
+4 |     if let 2 = a {}
+  |     ---------------
+  |
+
+//! > fixed
+fn main() {
+    let a = 2;
+    if a == 2 {}
+}
+
+//! > ==========================================================================
+
+//! > Struct Pattern Matching
+
+//! > cairo_code
+#[derive(Drop)]
+struct Template {
+    a: u8,
+    b: bool
+}
+
+fn main() {
+    let d = Template { a: 2, b: false };
+
+    if let Template { a: 2, b: false } = d {}
+}
+
+//! > diagnostics
+warning: Plugin diagnostic: `if let` pattern used for equatable value. Consider using a simple comparison `==` instead
+  --> lib.cairo:18:5
+   |
+18 |     if let Template { a: 2, b: false } = d {}
+   |     -----------------------------------------
+   |
+
+//! > fixed
+#[derive(Drop)]
+struct Template {
+    a: u8,
+    b: bool
+}
+
+fn main() {
+    let d = Template { a: 2, b: false };
+
+    if d == (Template { a: 2, b: false }) {}
+}
+
+//! > ==========================================================================
+
+//! > Enum Tuple Variant Pattern Matching with Tuple Struct
+
+//! > cairo_code
+struct Tuple {
+    first: u32,
+    second: u32
+}
+
+enum Enum {
+    TupleVariant(Tuple)
+}
+
+fn main() {
+    if let Enum::TupleVariant(32, 64) = e {}
+}
+
+//! > diagnostics
+warning: Plugin diagnostic: `if let` pattern used for equatable value. Consider using a simple comparison `==` instead
+  --> lib.cairo:20:5
+   |
+20 |     if let Enum::TupleVariant(32, 64) = e {}
+   |     ----------------------------------------
+   |
+error: Identifier not found.
+  --> lib.cairo:20:41
+   |
+20 |     if let Enum::TupleVariant(32, 64) = e {}
+   |                                         ^
+   |
+
+//! > fixed
+struct Tuple {
+    first: u32,
+    second: u32
+}
+
+enum Enum {
+    TupleVariant(Tuple)
+}
+
+fn main() {
+    if e == Enum::TupleVariant(32, 64 {}
+}
+
+//! > ==========================================================================
+
+//! > Enum Record Variant Pattern Matching
+
+//! > cairo_code
+enum Enum {
```

Comment:
the code should look like that if you want to unpack the enum the struct in the enum variant:
```rs
#[derive(Drop)]
struct B {
    a: i64,
    b: u32
}
#[derive(Drop)]
enum Enum {
    RecordVariant: B
}
fn main() {
    let e = Enum::RecordVariant(B {a: 1, b: 2});
    if let Enum::RecordVariant(B {a, b}) = e {}
}
```